src/assignment/components/AssignmentAdminFormEditable.tsx

```diff
@@ -0,0 +1,203 @@
+import { useGetQualityLabels } from '../../shared/hooks';
+import useTranslation from '../../shared/hooks/useTranslation';
+
+interface AssignmentAdminFormEditableProps {
+	assignment: Avo.Assignment.Assignment;
+	setAssignment: Dispatch<SetStateAction<Avo.Assignment.Assignment>>;
+	setValue: UseFormSetValue<Avo.Assignment.Assignment>;
+}
+
+const AssignmentAdminFormEditable: FC<AssignmentAdminFormEditableProps & UserProps> = ({
+	assignment,
+	user,
+	setAssignment,
+	setValue,
+}) => {
+	const { tText } = useTranslation();
+	const { data: qualityLabels, isLoading } = useGetQualityLabels();
+
+	const transformQualityLabelsToTagInfo = (labels: QualityLabel[]): TagInfo[] => {
+		return labels.map((label: QualityLabel) => ({
+			label: label.description,
+			value: label.value,
+		}));
+	};
+
+	const getAssignmentLabels = (): TagInfo[] => {
+		if (!qualityLabels) {
+			return [];
+		}
+		const labelIds = ((assignment.quality_labels || []) as Avo.Assignment.QualityLabel[]).map(
+			(item: any) => item.label
+		);
+		const filteredQualityLabels = qualityLabels.filter((qualityLabel) =>
+			labelIds.includes(qualityLabel.value)
+		);
+
+		return transformQualityLabelsToTagInfo(filteredQualityLabels);
+	};
+
+	const handleQualityLabelChange = (labels: TagInfo[]) => {
+		const newQualityLabels = labels.map(
+			(label) =>
+				({
+					assignment_id: assignment.id,
+					label: label.value,
+				} as Avo.Assignment.QualityLabel)
+		);
+
+		(setValue as any)('quality_labels', newQualityLabels, {
+			shouldDirty: true,
+			shouldTouch: true,
+		});
+		setAssignment((prev) => ({
+			...prev,
+			quality_labels: newQualityLabels,
+			blocks: (prev as Avo.Assignment.Assignment)?.blocks || [],
+		}));
+	};
+
+	const owner: PickerItem | undefined = assignment.profile
+		? {
+				label: `${assignment.profile.user.first_name} ${assignment.profile.user.last_name} (${assignment.profile.user.mail})`,
+				type: 'PROFILE',
+				value: assignment.profile.id,
+		  }
+		: undefined;
+
+	const handleOwnerChange = (newOwner: PickerItem | null) => {
+		if (isNil(newOwner)) {
+			return;
+		}
+		(setValue as any)('owner_profile_id', newOwner, {
+			shouldDirty: true,
+			shouldTouch: true,
+		});
+		setAssignment((prev) => ({
+			...prev,
+			owner_profile_id: newOwner.value,
+			blocks: (prev as Avo.Assignment.Assignment)?.blocks || [],
+		}));
+	};
+
+	return (
+		<Container mode="vertical">
+			<Container mode="horizontal">
+				<Form>
+					<Spacer margin="bottom">
+						<Grid>
+							<Column size="3-7">
+								<FormGroup label={tText('Laatst bewerkt door')}>
+									<TextInput
+										disabled
+										value={
+											getFullName(assignment.updated_by, true, false) || '-'
+										}
+									/>
+								</FormGroup>
+
+								<FormGroup label={tText('Aangepast op')}>
+									<TextInput
+										disabled
+										value={
+											assignment.updated_at
+												? formatTimestamp(assignment.updated_at)
+												: '-'
```

Comment:
formatTimestamp returns empty string if the date is null/undefined
and empty string is falsey so you can simplify this:
```suggestion
											formatTimestamp(assignment.updated_at) || '-'
```